embassy-rp/src/gpio.rs

```diff
@@ -265,6 +472,47 @@ impl<'d, T: Pin> Flex<'d, T> {
     pub fn toggle(&mut self) {
         unsafe { self.pin.sio_out().value_xor().write_value(self.bit()) }
     }
+
+    pub async fn wait_for_high<'a>(&mut self) {
+        if self.is_set_as_output() {
+            panic!("wait_for_high can't be called when in output mode");
```

Comment:
Don't do these checks, the whole point of Flex is it lets you do anything at the cost of some safety, so it's fine if it behaves weird if you use it in the wrong mode.

Reply:
Ah I see, I'll remove them, thanks!